### PR TITLE
Q6 complete

### PR DESCRIPTION
**Pull request by Venitious**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -103,3 +102,47 @@ describe('GET /api', () => {
         })
     });
 });
+
+describe('GET /api/reviews/:review_id/comments', () => {
+    it('should respond with all the comments asociated from that particular review  ', () => {
+        return request(app).get('/api/reviews/2/comments')
+        .expect(200)
+        .then((result) => {
+            const resultArr = result.body.comments
```
Comment (JVal-NC) [inline __tests__/app.test.js:111]: Can we check the comments are sorted as per the ticket, jest-sorted may come in handy.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -103,3 +102,47 @@ describe('GET /api', () => {
...
+    });
+    it('should respond with  a status code 200 & an empty array if passed an existing review_id which has not posted any comments', () => {
+        return request(app).get('/api/reviews/1/comments')
+        .expect(200)
+        .then((result) => {
+            const resultArr = result.body.comments
+            expect(resultArr.length).toBe(0)
+        })        
+    });
+    it('should respond with a 404 not found if passed a review_id which is currently not in use', () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:133]: Good error handling tests!

File: db/models.js [hunk 3]
```diff
@@ -45,6 +46,30 @@ exports.fetchReview = (reviewId) => {
...
+        .query(sqlQuery, sqlInsertion)
+        .then((result) => {
+            return result.rows
+        })
+    })
+}
+
+
+
+// const comments = result.rows[0];
```
Comment (JVal-NC) [inline db/models.js]: Remove commented out code from PRs please.

File: db/controllers.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
-const { fetchCategories, fetchReviews, fetchReview } = require("./models")
+const { fetchCategories, fetchReviews, fetchReview, fetchCommentsById } = require("./models")
 const fs = require('fs')
 const endpoints = require('../endpoints.json')
+const { doesCategoryExist } = require("./seeds/utils")
+
+
```
Comment (JVal-NC) [inline db/controllers.js:5]: Can you rename your controllers to more suitable RESTful API naming conventions, Get/Fetch, Post/Insert etc

File: db/models.js [hunk 5]
```diff
@@ -45,6 +46,30 @@ exports.fetchReview = (reviewId) => {
...
 
-exports.retrieveEndpoints = () => {
-    
-}
\ No newline at end of file
+exports.fetchCommentsById = (queryId) => {
+    const sqlInsertion = [queryId]
+    const sqlQuery = `SELECT * FROM comments
+    WHERE review_id = $1;`
+    return doesCategoryExist('reviews', 'review_id', queryId)
```
Comment (JVal-NC) [inline db/models.js:50]: I think using a util for this may be a bit unnecessary, makes it less readable.

**Review by JVal-NC — COMMENTED** [review]
Hey Geo, some initial comments to consider. Please submit another PR after and we'll do a full review for that one.

File: db/models.js (hunk 5 above)
Comment (Venitious) [reply db/models.js:50]: does it not make it re-usable in case we have to use it in future tasks?

File: __tests__/app.test.js [hunk 6]
```diff
@@ -109,6 +109,7 @@ describe('GET /api/reviews/:review_id/comments', () => {
         .expect(200)
         .then((result) => {
             const resultArr = result.body.comments
+            expect(resultArr).toBeSortedBy('created_at', {descending: true})
```
Comment (Augs0) [inline __tests__/app.test.js:112]: Great, exactly what we need :)

File: __tests__/app.test.js [hunk 7]
```diff
@@ -146,3 +147,23 @@ describe('GET /api/reviews/:review_id/comments', () => {
         })    
     });
 });
+
+// describe('POST /api/reviews/:review_id/comments', () => {
```
Comment (Augs0) [inline __tests__/app.test.js]: This commented out code shouldn't be here 👀

File: db/models.js [hunk 8]
```diff
@@ -60,16 +57,6 @@ exports.fetchCommentsById = (queryId) => {
...
-// 	  msg: `The id number ${queryId}, is not currently in use`,
-// 	});
-//   }
-// return user
-
-
-//basically the test we used in 'GET/api/reviews/:review_id' was to check that the rows were 0 and in that case return an error, the problem is that now rows containing 0 is a possiblity if that person has not made any comments. - Therefore I need to find a way to check that that review_id exists as a condition 
\ No newline at end of file
+exports.postCommentsById = (postRequest, queryId) => {
+    console.log(postRequest, queryId)
```
Comment (Augs0) [inline db/models.js]: This shouldn't be here either but I think you did go back and delete it afterwards :)

File: output.txt [hunk 9]
```diff
@@ -1,5 +1,8 @@
 You are now connected to database "nc_games_test" as user "geo".
- comment_id | body | review_id | author | votes | created_at 
-------------+------+-----------+--------+-------+------------
-(0 rows)
```
Comment (Augs0) [inline output.txt]: We don't need to commit playground files or any of their output

File: db/models.js (hunk 5 above)
Comment (Augs0) [reply db/models.js:50]: I understand what Joe is saying here but I think a util function is fine. I would reconsider your current util function though. Though it's intended to be reused with those three parameters allowing for different tables, columns and ids, your function can only currently query the reviews table due to the statement used inside. For what it's worth, I don't think we reuse this anywhere in the next tasks, but having a reusable function is a really great idea :)

File: __tests__/app.test.js [hunk 10]
```diff
@@ -133,8 +134,8 @@ describe('GET /api/reviews/:review_id/comments', () => {
         return request(app).get('/api/reviews/10000/comments')
         .expect(404)
         .then((result) => {
-            const errMsg = result.body.msg
-            expect(errMsg).toBe('The id number 10000, is not currently in use')
+            const errorMessage = result.body.msg
+            expect(errorMessage).toBe('The id number 10000, is not currently in use')
```
Comment (Augs0) [inline __tests__/app.test.js:138]: This is a good test but I would caution against overly specific error messages. They can unintentionally give bad actors info about your data. It's not a big deal here obviously as our database is available to see, but just something to keep in mind

**Review by Augs0 — APPROVED** [review]
This is great work Geo. I'm happy for you to merge once all your console logs and commented out code are cleaned up 😸

**Pull request merged**